compileopts/target.go

```diff
@@ -23,45 +23,46 @@ import (
-	CFlags           []string `json:"cflags"`
-	LDFlags          []string `json:"ldflags"`
-	LinkerScript     string   `json:"linkerscript"`
-	ExtraFiles       []string `json:"extra-files"`
-	RP2040BootPatch  *bool    `json:"rp2040-boot-patch"` // Patch RP2040 2nd stage bootloader checksum
-	Emulator         string   `json:"emulator"`
-	FlashCommand     string   `json:"flash-command"`
-	GDB              []string `json:"gdb"`
-	PortReset        string   `json:"flash-1200-bps-reset"`
-	SerialPort       []string `json:"serial-port"` // serial port IDs in the form "acm:vid:pid" or "usb:vid:pid"
-	FlashMethod      string   `json:"flash-method"`
-	FlashVolume      string   `json:"msd-volume-name"`
-	FlashFilename    string   `json:"msd-firmware-name"`
-	UF2FamilyID      string   `json:"uf2-family-id"`
-	BinaryFormat     string   `json:"binary-format"`
-	OpenOCDInterface string   `json:"openocd-interface"`
-	OpenOCDTarget    string   `json:"openocd-target"`
-	OpenOCDTransport string   `json:"openocd-transport"`
-	OpenOCDCommands  []string `json:"openocd-commands"`
-	OpenOCDVerify    *bool    `json:"openocd-verify"` // enable verify when flashing with openocd
-	JLinkDevice      string   `json:"jlink-device"`
-	CodeModel        string   `json:"code-model"`
-	RelocationModel  string   `json:"relocation-model"`
-	WasmAbi          string   `json:"wasm-abi"`
+	Inherits           []string `json:"inherits"`
+	Triple             string   `json:"llvm-target"`
+	CPU                string   `json:"cpu"`
+	Features           string   `json:"features"`
+	GOOS               string   `json:"goos"`
+	GOARCH             string   `json:"goarch"`
+	BuildTags          []string `json:"build-tags"`
+	GC                 string   `json:"gc"`
+	Scheduler          string   `json:"scheduler"`
+	Serial             string   `json:"serial"` // which serial output to use (uart, usb, none)
+	Linker             string   `json:"linker"`
+	RTLib              string   `json:"rtlib"` // compiler runtime library (libgcc, compiler-rt)
+	Libc               string   `json:"libc"`
+	AutoStackSize      *bool    `json:"automatic-stack-size"` // Determine stack size automatically at compile time.
+	DefaultStackSize   uint64   `json:"default-stack-size"`   // Default stack size if the size couldn't be determined at compile time.
+	CFlags             []string `json:"cflags"`
+	LDFlags            []string `json:"ldflags"`
+	LinkerScript       string   `json:"linkerscript"`
+	ExtraFiles         []string `json:"extra-files"`
+	RP2040BootPatch    *bool    `json:"rp2040-boot-patch"` // Patch RP2040 2nd stage bootloader checksum
+	Emulator           string   `json:"emulator"`
+	FlashCommand       string   `json:"flash-command"`
+	GDB                []string `json:"gdb"`
+	PortReset          string   `json:"flash-1200-bps-reset"`
+	PortResetIgnoreErr string   `json:"flash-1200-bps-reset-ignore-error"`
```

Comment:
Do we need this setting?  Might be best to always ignore errors?  (I can't really think of a reason where a developer would turn it off)

Reply:
I tried to reduce the impact on other targets as much as possible.
However, we certainly may not have to worry that much about it.

@deadprogram 
What do you think?

Reply:
Hmmm, it is hard to understand the error message when there are multiple options at the same time.


Before correction
```
# When two Wio Terminals are present
$ tinygo flash --target wioterminal --size short ./b1
   code    data     bss |   flash     ram
   7824     144    6260 |    7968    6404
error: multiple serial ports available - use -port flag, available ports are COM37, COM105
```

```
# When Wio Terminal does not exist
$ tinygo flash --target wioterminal --size short ./b1
   code    data     bss |   flash     ram
   7824     144    6260 |    7968    6404
error: multiple serial ports available - use -port flag, available ports are COM48, COM104
```


After correction (always ignore error)
```
# When two Wio Terminals are present
$ tinygo flash --target wioterminal --size short ./b1
   code    data     bss |   flash     ram
   7844     144    6260 |    7988    6404
error: failed to flash C:\Users\sago3\AppData\Local\Temp\tinygo1961070370\main.uf2: unable to locate a USB device to be flashed
```

```
# When Wio Terminal does not exist
$ tinygo flash --target wioterminal --size short ./b1
   code    data     bss |   flash     ram
   7844     144    6260 |    7988    6404
error: failed to flash C:\Users\sago3\AppData\Local\Temp\tinygo3745000764\main.uf2: unable to locate a USB device to be flashed
```

Reply:
Seems to me it would be better to either always ignore the PortReset failures, or else to find the root cause on the RP2040. Having this be "Yet Another Setting" is perhaps not the cleanest solution.

Reply:
The root cause is clear.

The rp2040 bootloader fails to open the serial port because it does not define a serial port.

A bootloader such as samd51 defines a serial port and ignores 1200bps-open.
Therefore, flash-1200-bps-reset works well with and without bootloader.

Reply:
PortReset failures are now ignored.
Error messages degrade in some cases, such as when multiple connections are made, but should be improved separately.